File: B3WM.Shared/Entity/Ticks2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace B3WM.Shared.Entity
{
    public class Ticks2
    {

        public int TrydID { get; set; }

        public DateTime Time { get; set; }
        public double Value { get; set; }
        public int Volume { get; set; }
        public Agents Buyer { get; set; }
        public Agents Seller { get; set; }
        public ActionType Starter { get; set; }
        public string Symbol { get; set; }

        public override string ToString()
        {
            return $"{TrydID}\t {Time.ToString("HH:mm:ss")}\t\t {Value}\t {Volume}\t {Buyer}\t\t\t {Seller}\t\t\t {Starter}";
        }

        public enum Agents
        {
            [Description("MAGLIANO S.A. CCVM")]
            [Display(Description = "MAGLIANO S.A. CCVM")]
            Magliano = 1,
            [Description("XP INVESTIMENTOS CCTVM S/A")]
            [Display(Description = "XP INVESTIMENTOS CCTVM S/A")]
            XP = 3,
            [Description("ALFA CCVM S.A.")]
            [Display(Description = "ALFA CCVM S.A.")]
            Alfa = 4,
            [Description("UBS BRASIL CCTVM S/A")]
            [Display(Description = "UBS BRASIL CCTVM S/A")]
            UBS = 8,
            [Description("MERRILL LYNCH S/A CTVM")]
            [Display(Description = "MERRILL LYNCH S/A CTVM")]
            Merrill = 13,
            [Description("GUIDE INVESTIMENTOS S.A. CV")]
            [Display(Description = "GUIDE INVESTIMENTOS S.A. CV")]
            Guide = 15,
            [Description("J. P. MORGAN CCVM S.A.")]
            [Display(Description = "J. P. MORGAN CCVM S.A.")]
            JP = 16,
            [Description("BOCOM BBM CCVM S/A")]
            [Display(Description = "BOCOM BBM CCVM S/A")]
            BOCOM = 18,
            [Description("VOTORANTIM ASSET MANAG. DTVM")]
            [Display(Description = "VOTORANTIM ASSET MANAG. DTVM")]
            Votorantim = 21,
            [Description("NECTON INVESTIMENTOS S.A. CVMC")]
            [Display(Description = "NECTON INVESTIMENTOS S.A. CVMC")]
            Necton = 23,

            [Description("SANTANDER CCVM S/A")]
            [Display(Description = "SANTANDER CCVM S/A")]
            Santander = 27,

            [Description("UNILETRA CCTVM S.A.")]
            [Display(Description = "UNILETRA CCTVM S.A.")]
            Uniletra = 29,

            [Description("LEROSA S.A. CVC")]
            [Display(Description = "LEROSA S.A. CVC")]
            Lerosa = 33,

            [Description("UM INVESTIMENTOS S.A. CTVM")]
            [Display(Description = "UM INVESTIMENTOS S.A. CTVM")]
            Um = 37,

            [Description("AGORA CTVM S/A")]
            [Display(Description = "AGORA CTVM S/A")]
            Agora = 39,

            [Description("MORGAN STANLEY CTVM S/A")]
            [Display(Description = "MORGAN STANLEY CTVM S/A")]
            Morgan = 40,

            [Description("ING CCT S/A")]
            [Display(Description = "ING CCT S/A")]
            Ing = 41,

            [Description("CREDIT SUISSE BRASIL S.A. CTVM")]
            [Display(Description = "CREDIT SUISSE BRASIL S.A. CTVM")]
            Credit = 45,

            [Description("SOCOPA SC PAULISTA S.A.")]
            [Display(Description = "SOCOPA SC PAULISTA S.A.")]
            Socopa = 58,

            [Description("SAFRA CVC LTDA.")]
            [Display(Description = "SAFRA CVC LTDA.")]
            Safra = 59,

            [Description("NOVINVEST CVM LTDA.")]
            [Display(Description = "NOVINVEST CVM LTDA.")]
            Novinvest = 63,

            [Description("BRADESCO S/A CTVM")]
            [Display(Description = "BRADESCO S/A CTVM")]
            Bradesco = 72,

            [Description("COINVALORES CCVM LTDA.")]
            [Display(Description = "COINVALORES CCVM LTDA.")]
            Coinvalores = 74,

            [Description("CITIGROUP GMB CCTVM S.A.")]
            [Display(Description = "CITIGROUP GMB CCTVM S.A.")]
            Citigroup = 77,

            [Description("MAXIMA S/A CTVM")]
            [Display(Description = "MAXIMA S/A CTVM")]
            Maxima = 83,

            [Description("BTG PACTUAL CTVM S.A.")]
            [Display(Description = "BTG PACTUAL CTVM S.A.")]
            BTG = 85,

            [Description("CM CAPITAL MARKETS CCTVM LTDA")]
            [Display(Description = "CM CAPITAL MARKETS CCTVM LTDA")]
            Capital = 88,

            [Description("NUINVEST – TITULO CV S.A.")]
            [Display(Description = "NUINVEST – TITULO CV S.A.")]
            NuInvest = 90,

            [Description("RENASCENCA DTVM LTDA.")]
            [Display(Description = "RENASCENCA DTVM LTDA.")]
            Renascenca = 92,

            [Description("NOVA FUTURA CTVM LTDA")]
            [Display(Description = "NOVA FUTURA CTVM LTDA")]
            Nova_Futura = 93,

            [Description("MERC. DO BRASIL COR. S.A. CTVM")]
            [Display(Description = "MERC. DO BRASIL COR. S.A. CTVM")]
            Mercantil = 106	,

            [Description("TERRA INVESTIMENTOS DTVM LTDA")]
            [Display(Description = "TERRA INVESTIMENTOS DTVM LTDA")]
            Terra = 107	,

            [Description("SLW CVC LTDA.")]
            [Display(Description = "SLW CVC LTDA.")]
            SLW = 110	,

            [Description("ITAU CV S/A")]
            [Display(Description = "ITAU CV S/A")]
            Itau = 114	,

            [Description("H.COMMCOR DTVM LTDA")]
            [Display(Description = "H.COMMCOR DTVM LTDA")]
            HCOMMCOR = 115,

            [Description("GENIAL INSTITUCIONAL CCTVM S.A")]
            [Display(Description = "GENIAL INSTITUCIONAL CCTVM S.A")]
            Genial = 120,

            [Description("BGC LIQUIDEZ DTVM")]
            [Display(Description = "BGC LIQUIDEZ DTVM")]
            BGC_Liquidez = 122,

            [Description("TULLETT PREBON")]
            [Display(Description = "TULLETT PREBON")]
            Tullet = 127,

            [Description("PLANNER CV S.A")]
            [Display(Description = "PLANNER CV S.A")]
            Planner = 129,

            [Description("FATOR S.A. CV")]
            [Display(Description = "FATOR S.A. CV")]
            Fator = 131,

            [Description("DIBRAN DTVM LTDA")]
            [Display(Description = "DIBRAN DTVM LTDA")]
            Dibran = 133,

            [Description("ATIVA INVESTIMENTOS S.A. CTCV")]
            [Display(Description = "ATIVA INVESTIMENTOS S.A. CTCV")]
            Ativa = 147,

            [Description("BANRISUL S/A CVMC")]
            [Display(Description = "BANRISUL S/A CVMC")]
            Banrisul = 172,

            [Description("GENIAL INVESTIMENTOS CVM S.A.")]
            [Display(Description = "GENIAL INVESTIMENTOS CVM S.A.")]
            Genial_Invest = 173,

            [Description("ELITE CCVM LTDA.")]
            [Display(Description = "ELITE CCVM LTDA.")]
            Elite = 174,

            [Description("SOLIDUS S/A CCVM")]
            [Display(Description = "SOLIDUS S/A CCVM")]
            Solidus = 177,

            [Description("MUNDINVEST S.A. CCVM")]
            [Display(Description = "MUNDINVEST S.A. CCVM")]
            Mundinvest = 181,

            [Description("CORRETORA GERAL DE VC LTDA")]
            [Display(Description = "CORRETORA GERAL DE VC LTDA")]
            Geral = 186,

            [Description("SITA SCCVM S.A.")]
            [Display(Description = "SITA SCCVM S.A.")]
            Sita = 187,

            [Description("ELLIOT WARREN.")]
            [Display(Description = "ELLIOT WARREN.")]
            Elliot_Warren = 190,

            [Description("SENSO CCVM S.A.")]
            [Display(Description = "SENSO CCVM S.A.")]
            Senso = 191,

            [Description("AMARIL FRANKLIN CTV LTDA.")]
            [Display(Description = "AMARIL FRANKLIN CTV LTDA.")]
            Amaril = 226,

            [Description("CODEPE CV E CAMBIO S/A")]
            [Display(Description = "CODEPE CV E CAMBIO S/A")]
            Codepe = 234,

            [Description("GOLDMAN SACHS DO BRASIL CTVM")]
            [Display(Description = "GOLDMAN SACHS DO BRASIL CTVM")]
            Goldman = 238,

            [Description("BANCO BNP PARIBAS BRASIL S/A")]
            [Display(Description = "BANCO BNP PARIBAS BRASIL S/A")]
            Banco_BNP = 251,

            [Description("MIRAE ASSET WEALTH MANAGEMENT")]
            [Display(Description = "MIRAE ASSET WEALTH MANAGEMENT")]
            Mirae = 262,

            [Description("CLEAR CORRETORA – Grupo XP")]
            [Display(Description = "CLEAR CORRETORA – Grupo XP")]
            Clear = 308,

            [Description("PAGINVEST")]
            [Display(Description = "PAGINVEST")]
            PAGINVEST = 357,

            [Description("BANCO DAYCOVAL")]
            [Display(Description = "BANCO DAYCOVAL")]
            Daycoval = 359,

            [Description("RICO INVESTIMENTOS – Grupo XP")]
            [Display(Description = "RICO INVESTIMENTOS – Grupo XP")]
            Rico = 386,
            [Description("BANCO OURINVEST")]
            [Display(Description = "BANCO OURINVEST")]
            Banco_Ourinvest = 442,
            [Description("BANCO MODAL")]
            [Display(Description = "BANCO MODAL")]
            Banco_Modal = 683,

            [Description("ABN AMRO CLEARING CTVM LTDA")]
            [Display(Description = "ABN AMRO CLEARING CTVM LTDA")]
            ABN_AMRO = 688,

            [Description("DILLON S.A. DTVM")]
            [Display(Description = "DILLON S.A. DTVM")]
            Dillon = 711,

            [Description("BB GESTAO DE RECURSOS DTVM S/A")]
            [Display(Description = "BB GESTAO DE RECURSOS DTVM S/A")]
            BB_Recursos = 713,

            [Description("ICAP DO BRASIL CTVM LTDA")]
            [Display(Description = "ICAP DO BRASIL CTVM LTDA")]
            Icap = 735,

            [Description("LEV DTVM LTDA")]
            [Display(Description = "LEV DTVM LTDA")]
            LEV_DTVM = 746,

            [Description("BB BANCO DE INVESTIMENTO S/A")]
            [Display(Description = "BB BANCO DE INVESTIMENTO S/A")]
            BB = 820,

            [Description("ADVALOR DTVM LTDA")]
            [Display(Description = "ADVALOR DTVM LTDA")]
            Advalor = 979,

            [Description("RB CAPITAL INVESTIMENTOS DTVM")]
            [Display(Description = "RB CAPITAL INVESTIMENTOS DTVM")]
            RB_Capital = 1089,

            [Description("INTER DTVM LTDA")]
            [Display(Description = "INTER DTVM LTDA")]
            Inter = 1099,

            [Description("Ourinvest DTVM S.A.")]
            [Display(Description = "Ourinvest DTVM S.A.")]
            Ourinvest = 1106,

            [Description("BANCO CITIBANK")]
            [Display(Description = "BANCO CITIBANK")]
            Citibank = 1116,

            [Description("INTL FCStone DTVM Ltda.")]
            [Display(Description = "INTL FCStone DTVM Ltda.")]
            Intl = 1130,

            [Description("CAIXA ECONOMICA FEDERAL")]
            [Display(Description = "CAIXA ECONOMICA FEDERAL")]
            Caixa = 1570,

            [Description("IDEAL CTVM SA")]
            [Display(Description = "IDEAL CTVM SA")]
            Ideal = 1618,

            [Description("MODAL DTVM LTDA")]
            [Display(Description = "MODAL DTVM LTDA")]
            Modal = 1982,

            [Description("BCO FIBRA")]
            [Display(Description = "BCO FIBRA")]
            BCO_Fibra = 2197,

            [Description("ORLA DTVM S/A")]
            [Display(Description = "ORLA DTVM S/A")]
            Orla = 2379,

            [Description("POSITIVA CTVM S/A")]
            [Display(Description = "POSITIVA CTVM S/A")]
            Positiva = 2492,

            [Description("SANTANDER SECURITIES SERVICES")]
            [Display(Description = "SANTANDER SECURITIES SERVICES")]
            Santander_Securities = 2570,
            [Description("LLA DTVM LTDA")]
            [Display(Description = "LLA DTVM LTDA")]
            LLA = 2640,
            [Description("BANESTES DTVM S/A")]
            [Display(Description = "BANESTES DTVM S/A")]
            Banestes = 3112,

            [Description("RIO BRAVO INVEST S.A. DTVM")]
            [Display(Description = "RIO BRAVO INVEST S.A. DTVM")]
            Rio_Bravo = 3371,

            [Description("ORAMA DTVM S.A.")]
            [Display(Description = "ORAMA DTVM S.A.")]
            Orama = 3701,

            [Description("RJI CTVM LTDA")]
            [Display(Description = "RJI CTVM LTDA")]
            RJI = 3762,

            [Description("BANCO ANDBANK (BRASIL) S.A.")]
            [Display(Description = "BANCO ANDBANK (BRASIL) S.A.")]
            AndBank = 4002,

            [Description("BS2 DTVM S/A")]
            [Display(Description = "BS2 DTVM S/A")]
            BS2 = 4015,

            [Description("TORO CTVM LTDA.")]
            [Display(Description = "TORO CTVM LTDA.")]
            Toro = 4090,

            [Description("C6 CTVM LTDA")]
            [Display(Description = "C6 CTVM LTDA")]
            C6 = 6003
        }
        public enum ActionType
        {
            [Display(Description = "Compra")]
            [Description("Compra")]
            Buy = 1,
            [Display(Description = "Venda")]
            [Description("Venda")]
            Sale = 2,
            [Display(Description = "Leilão")]
            [Description("Leilão")]
            Auction = 3,
            [Display(Description = "Direto")]
            [Description("Direto")]
            Cross = 4,
            [Display(Description = "RLP")]
            [Description("RLP")]
            RLP = 5,
        }

        //public static bool CheckAnomaly(IDictionary<Agents, int> balanceByAgents, double multiplierSame, double multiplierOther, ActionType starter)
        //{
        //    try
        //    {
        //        if (balanceByAgents == null || balanceByAgents.Count == 0)
        //            return false;

        //        var maxs = balanceByAgents.OrderByDescending(q => q.Value).Take(2).ToList();
        //        var mins = balanceByAgents.OrderBy(q => q.Value).Take(2).ToList();

        //        if (starter == ActionType.Buy)
        //        {
        //            if (maxs[0].Value * multiplierSame > maxs[1].Value &&
        //                maxs[0].Value * multiplierOther > Math.Abs(mins[0].Value))
        //                return true;
        //        }
        //        else if (starter == ActionType.Sale)
        //        {
        //            if (mins[0].Value * multiplierSame < mins[1].Value &&
        //                Math.Abs(mins[0].Value * multiplierOther) > maxs[0].Value)
        //                return true;
        //        }
        //    }catch(Exception expt)
        //    {
        //        Console.WriteLine(expt.Message);
        //        Console.WriteLine("CheckAnomaly");
        //    }


        //    return false;
        //}
    }
}

File: B3WM.Flutter/lib/models/ticks2.dart
import 'package:flutter/foundation.dart';

enum ActionType {
  buy(1, 'Compra'),
  sale(2, 'Venda'),
  auction(3, 'Leilão'),
  cross(4, 'Direto'),
  rlp(5, 'RLP');

  final int value;
  final String description;
  const ActionType(this.value, this.description);

  static ActionType fromValue(int value) {
    return ActionType.values.firstWhere(
      (e) => e.value == value,
      orElse: () => ActionType.buy,
    );
  }
}

@immutable
class Agents {
  final int value;
  final String description;

  const Agents._(this.value, this.description);

  static const magliano = Agents._(1, 'MAGLIANO S.A. CCVM');
  static const xp = Agents._(3, 'XP INVESTIMENTOS CCTVM S/A');
  static const alfa = Agents._(4, 'ALFA CCVM S.A.');
  static const ubs = Agents._(8, 'UBS BRASIL CCTVM S/A');
  static const merrill = Agents._(13, 'MERRILL LYNCH S/A CTVM');
  static const guide = Agents._(15, 'GUIDE INVESTIMENTOS S.A. CV');
  static const jp = Agents._(16, 'J. P. MORGAN CCVM S.A.');
  static const bocom = Agents._(18, 'BOCOM BBM CCVM S/A');
  static const votorantim = Agents._(21, 'VOTORANTIM ASSET MANAG. DTVM');
  static const necton = Agents._(23, 'NECTON INVESTIMENTOS S.A. CVMC');
  static const santander = Agents._(27, 'SANTANDER CCVM S/A');
  static const uniletra = Agents._(29, 'UNILETRA CCTVM S.A.');
  static const lerosa = Agents._(33, 'LEROSA S.A. CVC');
  static const um = Agents._(37, 'UM INVESTIMENTOS S.A. CTVM');
  static const agora = Agents._(39, 'AGORA CTVM S/A');
  static const morgan = Agents._(40, 'MORGAN STANLEY CTVM S/A');
  static const ing = Agents._(41, 'ING CCT S/A');
  static const credit = Agents._(45, 'CREDIT SUISSE BRASIL S.A. CTVM');
  static const socopa = Agents._(58, 'SOCOPA SC PAULISTA S.A.');
  static const safra = Agents._(59, 'SAFRA CVC LTDA.');
  static const novinvest = Agents._(63, 'NOVINVEST CVM LTDA.');
  static const bradesco = Agents._(72, 'BRADESCO S/A CTVM');
  static const coinvalores = Agents._(74, 'COINVALORES CCVM LTDA.');
  static const citigroup = Agents._(77, 'CITIGROUP GMB CCTVM S.A.');
  static const maxima = Agents._(83, 'MAXIMA S/A CTVM');
  static const btg = Agents._(85, 'BTG PACTUAL CTVM S.A.');
  static const capital = Agents._(88, 'CM CAPITAL MARKETS CCTVM LTDA');
  static const nuInvest = Agents._(90, 'NUINVEST – TITULO CV S.A.');
  static const renascenca = Agents._(92, 'RENASCENCA DTVM LTDA.');
  static const novaFutura = Agents._(93, 'NOVA FUTURA CTVM LTDA');
  static const mercantil = Agents._(106, 'MERC. DO BRASIL COR. S.A. CTVM');
  static const terra = Agents._(107, 'TERRA INVESTIMENTOS DTVM LTDA');
  static const slw = Agents._(110, 'SLW CVC LTDA.');
  static const itau = Agents._(114, 'ITAU CV S/A');
  static const hcommcor = Agents._(115, 'H.COMMCOR DTVM LTDA');
  static const genial = Agents._(120, 'GENIAL INSTITUCIONAL CCTVM S.A');
  static const bgcLiquidez = Agents._(122, 'BGC LIQUIDEZ DTVM');
  static const tullet = Agents._(127, 'TULLETT PREBON');
  static const planner = Agents._(129, 'PLANNER CV S.A');
  static const fator = Agents._(131, 'FATOR S.A. CV');
  static const dibran = Agents._(133, 'DIBRAN DTVM LTDA');
  static const ativa = Agents._(147, 'ATIVA INVESTIMENTOS S.A. CTCV');
  static const banrisul = Agents._(172, 'BANRISUL S/A CVMC');
  static const genialInvest = Agents._(173, 'GENIAL INVESTIMENTOS CVM S.A.');
  static const elite = Agents._(174, 'ELITE CCVM LTDA.');
  static const solidus = Agents._(177, 'SOLIDUS S/A CCVM');
  static const mundinvest = Agents._(181, 'MUNDINVEST S.A. CCVM');
  static const geral = Agents._(186, 'CORRETORA GERAL DE VC LTDA');
  static const sita = Agents._(187, 'SITA SCCVM S.A.');
  static const elliotWarren = Agents._(190, 'ELLIOT WARREN.');
  static const senso = Agents._(191, 'SENSO CCVM S.A.');
  static const amaril = Agents._(226, 'AMARIL FRANKLIN CTV LTDA.');
  static const codepe = Agents._(234, 'CODEPE CV E CAMBIO S/A');
  static const goldman = Agents._(238, 'GOLDMAN SACHS DO BRASIL CTVM');
  static const bancoBnp = Agents._(251, 'BANCO BNP PARIBAS BRASIL S/A');
  static const mirae = Agents._(262, 'MIRAE ASSET WEALTH MANAGEMENT');
  static const clear = Agents._(308, 'CLEAR CORRETORA – Grupo XP');
  static const paginvest = Agents._(357, 'PAGINVEST');
  static const daycoval = Agents._(359, 'BANCO DAYCOVAL');
  static const rico = Agents._(386, 'RICO INVESTIMENTOS – Grupo XP');
  static const bancoOurinvest = Agents._(442, 'BANCO OURINVEST');
  static const bancoModal = Agents._(683, 'BANCO MODAL');
  static const abnAmro = Agents._(688, 'ABN AMRO CLEARING CTVM LTDA');
  static const dillon = Agents._(711, 'DILLON S.A. DTVM');
  static const bbRecursos = Agents._(713, 'BB GESTAO DE RECURSOS DTVM S/A');
  static const icap = Agents._(735, 'ICAP DO BRASIL CTVM LTDA');
  static const levDtvM = Agents._(746, 'LEV DTVM LTDA');
  static const bb = Agents._(820, 'BB BANCO DE INVESTIMENTO S/A');
  static const advalor = Agents._(979, 'ADVALOR DTVM LTDA');
  static const rbCapital = Agents._(1089, 'RB CAPITAL INVESTIMENTOS DTVM');
  static const inter = Agents._(1099, 'INTER DTVM LTDA');
  static const ourinvest = Agents._(1106, 'Ourinvest DTVM S.A.');
  static const citibank = Agents._(1116, 'BANCO CITIBANK');
  static const intl = Agents._(1130, 'INTL FCStone DTVM Ltda.');
  static const caixa = Agents._(1570, 'CAIXA ECONOMICA FEDERAL');
  static const ideal = Agents._(1618, 'IDEAL CTVM SA');
  static const modal = Agents._(1982, 'MODAL DTVM LTDA');
  static const bcoFibra = Agents._(2197, 'BCO FIBRA');
  static const orla = Agents._(2379, 'ORLA DTVM S/A');
  static const positiva = Agents._(2492, 'POSITIVA CTVM S/A');
  static const santanderSecurities = Agents._(2570, 'SANTANDER SECURITIES SERVICES');
  static const lla = Agents._(2640, 'LLA DTVM LTDA');
  static const banestes = Agents._(3112, 'BANESTES DTVM S/A');
  static const rioBravo = Agents._(3371, 'RIO BRAVO INVEST S.A. DTVM');
  static const orama = Agents._(3701, 'ORAMA DTVM S.A.');
  static const rji = Agents._(3762, 'RJI CTVM LTDA');
  static const andBank = Agents._(4002, 'BANCO ANDBANK (BRASIL) S.A.');
  static const bs2 = Agents._(4015, 'BS2 DTVM S/A');
  static const toro = Agents._(4090, 'TORO CTVM LTDA.');
  static const c6 = Agents._(6003, 'C6 CTVM LTDA');

  static final Map<int, Agents> _byValue = {
    for (final a in _all) a.value: a,
  };

  static final List<Agents> _all = [
    magliano, xp, alfa, ubs, merrill, guide, jp, bocom, votorantim,
    necton, santander, uniletra, lerosa, um, agora, morgan, ing, credit,
    socopa, safra, novinvest, bradesco, coinvalores, citigroup, maxima,
    btg, capital, nuInvest, renascenca, novaFutura, mercantil, terra,
    slw, itau, hcommcor, genial, bgcLiquidez, tullet, planner, fator,
    dibran, ativa, banrisul, genialInvest, elite, solidus, mundinvest,
    geral, sita, elliotWarren, senso, amaril, codepe, goldman, bancoBnp,
    mirae, clear, paginvest, daycoval, rico, bancoOurinvest, bancoModal,
    abnAmro, dillon, bbRecursos, icap, levDtvM, bb, advalor, rbCapital,
    inter, ourinvest, citibank, intl, caixa, ideal, modal, bcoFibra,
    orla, positiva, santanderSecurities, lla, banestes, rioBravo, orama,
    rji, andBank, bs2, toro, c6,
  ];

  static Agents? fromValue(int value) => _byValue[value];

  @override
  bool operator ==(Object other) =>
      identical(this, other) || (other is Agents && value == other.value);

  @override
  int get hashCode => value.hashCode;
}
